### PR TITLE
`jsdoc/require-param-type` linting

### PR DESCRIPTION
**Pull request by arindam1993**
Depends on https://github.com/mapbox/mapbox-gl-js/pull/10864 for the `object -> Object` config change

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -1389,7 +1389,7 @@ class Map extends Camera {
      * style and the given style are different in any way, the map renderer will force a full update, removing the current style and building
      * the given one from scratch.
      *
-     * @param style A JSON object conforming to the schema described in the
+     * @param {Object | string| null} style A JSON object conforming to the schema described in the
      *   [Mapbox Style Specification](https://mapbox.com/mapbox-gl-style-spec/), or a URL to such JSON.
```
Comment (domlet) [inline src/ui/map.js:1393]:
```suggestion
     *   [Mapbox Style Specification](https://docs.mapbox.com/mapbox-gl-js/style-spec/), or a URL to such JSON.
```

**Review by domlet — APPROVED** [review]
`require-param-type` is a great improvement to the GL JS docs!

File: src/source/geojson_source.js [hunk 2]
```diff
@@ -185,8 +185,8 @@ class GeoJSONSource extends Evented implements Source {
     /**
      * For clustered sources, fetches the zoom at which the given cluster expands.
      *
-     * @param clusterId The value of the cluster's `cluster_id` property.
-     * @param callback A callback to be called when the zoom value is retrieved (`(error, zoom) => { ... }`).
+     * @param {number} clusterId The value of the cluster's `cluster_id` property.
+     * @param {Callback<number>} callback A callback to be called when the zoom value is retrieved (`(error, zoom) => { ... }`).
```
Comment (karimnaaji) [inline src/source/geojson_source.js:189]: Should this eventually mention the error in the callback? e.g. Callback<Error, number>?

File: src/source/geojson_source.js [hunk 3]
```diff
@@ -220,8 +220,8 @@ class GeoJSONSource extends Evented implements Source {
     /**
      * For clustered sources, fetches the children of the given cluster on the next zoom level (as an array of GeoJSON features).
      *
-     * @param clusterId The value of the cluster's `cluster_id` property.
-     * @param callback A callback to be called when the features are retrieved (`(error, features) => { ... }`).
+     * @param {number} clusterId The value of the cluster's `cluster_id` property.
+     * @param {Callback<Array<GeoJSONFeature>>} callback A callback to be called when the features are retrieved (`(error, features) => { ... }`).
```
Comment (karimnaaji) [inline src/source/geojson_source.js:224]: Same question as above for the error type, and for the one below line 254.

**Review by karimnaaji — APPROVED** [review]
Looks good to me, one question about the Callback syntax for the mention of the Error param.

**Comment by HeyStenson** [comment]
@arindam1993 is there anything else that needs to happen here? Or is it ready to merge into the main testing branch? Let me know if I can help at all!

**Comment by arindam1993** [comment]
@HeyStenson its ready to go!

**Pull request merged**